Gopkg.toml

```diff
@@ -6,42 +6,41 @@ required = [
   "k8s.io/code-generator/cmd/client-gen",
   "k8s.io/code-generator/cmd/lister-gen",
   "k8s.io/code-generator/cmd/informer-gen",
-  "k8s.io/code-generator/cmd/openapi-gen",
   "k8s.io/gengo/args",
 ]
 
 [[override]]
   name = "k8s.io/code-generator"
-  # revision for tag "kubernetes-1.12.3"
-  revision = "3dcf91f64f638563e5106f21f50c31fa361c918d"
+  # revision for tag "kubernetes-1.14.0"
+  revision = "50b561225d70b3eb79a1faafd3dfe7b1a62cbe73"
 
 [[override]]
   name = "k8s.io/api"
-  # revision for tag "kubernetes-1.12.3"
-  revision = "b503174bad5991eb66f18247f52e41c3258f6348"
+  # revision for tag "kubernetes-1.14.0"
+  revision = "40a48860b5abbba9aa891b02b32da429b08d96a0"
 
 [[override]]
   name = "k8s.io/apiextensions-apiserver"
-  # revision for tag "kubernetes-1.12.3"
-  revision = "0cd23ebeb6882bd1cdc2cb15fc7b2d72e8a86a5b"
+  # revision for tag "kubernetes-1.14.0"
+  revision = "53c4693659ed354d76121458fb819202dd1635fa"
 
 [[override]]
   name = "k8s.io/apimachinery"
-  # revision for tag "kubernetes-1.12.3"
-  revision = "eddba98df674a16931d2d4ba75edc3a389bf633a"
+  # revision for tag "kubernetes-1.14.0"
+  revision = "d7deff9243b165ee192f5551710ea4285dcfd615"
 
 [[override]]
   name = "k8s.io/client-go"
-  # revision for tag "kubernetes-1.12.3"
-  revision = "d082d5923d3cc0bfbb066ee5fbdea3d0ca79acf8"
+  # revision for tag "kubernetes-1.14.0"
+  revision = "6ee68ca5fd8355d024d02f9db0b3b667e8357a0f"
 
 [[override]]
   name = "github.com/coreos/prometheus-operator"
   version = "=v0.26.0"
 
 [[override]]
   name = "sigs.k8s.io/controller-runtime"
-  version = "=v0.1.8"
+  version = "=v0.2.0-beta.1"
```

Comment:
don't bump this yet, this version is definitely not ready to go.

Reply:
So yeah, this tripped me up.
The issue is that 0.2.0-beta.1 is the tag for controller-runtime to bump to Kubernetes 1.14.
Without pinning to that, we hit this on compile time:

```
# github.com/openshift/cluster-network-operator/vendor/sigs.k8s.io/controller-runtime/pkg/leaderelection
vendor/sigs.k8s.io/controller-runtime/pkg/leaderelection/leader_election.go:83:25: not enough arguments in call to resourcelock.New
        have (string, string, string, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, resourcelock.ResourceLockConfig)
        want (string, string, string, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/coordination/v1".CoordinationV1Interface, resourcelock.ResourceLockConfig)
```

Check pkg/leaderelection/leader_election.go.
https://github.com/kubernetes-sigs/controller-runtime/commit/71cf6f6cf312e9111fc0ff1b411e19b91726359d#diff-2590281733c8f856c7a8fb319d5d3cc5R87

So I wonder if we should wait for controller-runtime to release a version that is non-beta, then we change the pin.
Reverting to 0.1.8 is a no-go.

Reply:
Ugh. @LiliC any suggestions?

Reply:
Yes, we have a PR out in operator-sdk to switch to controller-runtime `0.2.0-x`, we haven't merged it yet as they are still in beta (from what I spoke with the maintainer they should be cutting the stable release soon if no bugs are found in testing). There are a lot of changes there, you can see in https://github.com/operator-framework/operator-sdk/pull/1388 PR if that is of any help. Sadly they did not release an intermediate version with bumping k8s to `1.14`.

I would agree to the following as we have things ready:

> So I wonder if we should wait for controller-runtime to release a version that is non-beta, then we change the pin.